## Stat 159 Final Project: Diabetes 
### By Bruce, Sam, Duy-Anh, Donghoon

In [22]:
#import statements
import numpy as np 
import pandas as pd
import statsmodels.formula.api as sm 

In [23]:
#datasets
diabetes_012_dataset = pd.read_csv("../data/diabetes_012_health_indicators_BRFSS2015.csv")
diabetes_binary_dataset =  pd.read_csv("../data/diabetes_binary_health_indicators_BRFSS2015.csv")

### Dataset Description 

In [24]:
#dataset with diabetes status 
diabetes_012_dataset

Diabetes_012  HighBP  HighChol  CholCheck   BMI  Smoker  Stroke  \
0                0.0     1.0       1.0        1.0  40.0     1.0     0.0   
1                0.0     0.0       0.0        0.0  25.0     1.0     0.0   
2                0.0     1.0       1.0        1.0  28.0     0.0     0.0   
3                0.0     1.0       0.0        1.0  27.0     0.0     0.0   
4                0.0     1.0       1.0        1.0  24.0     0.0     0.0   
...              ...     ...       ...        ...   ...     ...     ...   
253675           0.0     1.0       1.0        1.0  45.0     0.0     0.0   
253676           2.0     1.0       1.0        1.0  18.0     0.0     0.0   
253677           0.0     0.0       0.0        1.0  28.0     0.0     0.0   
253678           0.0     1.0       0.0        1.0  23.0     0.0     0.0   
253679           2.0     1.0       1.0        1.0  25.0     0.0     0.0   

        HeartDiseaseorAttack  PhysActivity  Fruits  ...  AnyHealthcare  \
0                        0.0           0.0     0.0  ...            1.0   
1                        0.0           1.0     0.0  ...            0.0   
2                        0.0           0.0     1.0  ...            1.0   
3                        0.0           1.0     1.0  ...            1.0   
4                        0.0           1.0     1.0  ...            1.0   
...                      ...           ...     ...  ...            ...   
253675                   0.0           0.0     1.0  ...            1.0   
253676                   0.0           0.0     0.0  ...            1.0   
253677                   0.0           1.0     1.0  ...            1.0   
253678                   0.0           0.0     1.0  ...            1.0   
253679                   1.0           1.0     1.0  ...            1.0   

        NoDocbcCost  GenHlth  MentHlth  PhysHlth  DiffWalk  Sex   Age  \
0               0.0      5.0      18.0      15.0       1.0  0.0   9.0   
1               1.0      3.0       0.0       0.0       0.0  0.0   7.0   
2               1.0      5.0      30.0      30.0       1.0  0.0   9.0   
3               0.0      2.0       0.0       0.0       0.0  0.0  11.0   
4               0.0      2.0       3.0       0.0       0.0  0.0  11.0   
...             ...      ...       ...       ...       ...  ...   ...   
253675          0.0      3.0       0.0       5.0       0.0  1.0   5.0   
253676          0.0      4.0       0.0       0.0       1.0  0.0  11.0   
253677          0.0      1.0       0.0       0.0       0.0  0.0   2.0   
253678          0.0      3.0       0.0       0.0       0.0  1.0   7.0   
253679          0.0      2.0       0.0       0.0       0.0  0.0   9.0   

        Education  Income  
0             4.0     3.0  
1             6.0     1.0  
2             4.0     8.0  
3             3.0     6.0  
4             5.0     4.0  
...           ...     ...  
253675        6.0     7.0  
253676        2.0     4.0  
253677        5.0     2.0  
253678        5.0     1.0  
253679        6.0     2.0  

[253680 rows x 22 columns]

In this dataset, there are 22 columns in total: 7 numeric columns, 14 binary columns (dummy), and the dependent column of interest (Diabetes_012). <br/><br/>

'Diabetes_012':  indicates that 0 is for no diabetes or only during pregnancy, 1 is for prediabetes, and 2 is for diabetes. <br/>
'HighBP': indicates that 0 is for low blood pressure, 1 is for high blood pressure <br/>
'HighChol': indicates that 0 is for low cholesterol level, 1 is for high cholesterol level <br/>
'CholCheck': indicates that 0 is for no cholesterol check within 5 years, 1 is for cholesterol check within 5 years <br/>
'BMI': Body mass index level <br/>
'Smoker': indicates that 0 is for never had more than 100 cigarettes, 1 is for had more than 100 cigarettes <br/>
'Stroke': indicates that 0 is for never had a stroke, 1 is for had a stroke <br/>
'HeartDiseaseorAttack': indicates that 0 is for not having coronary heart disease, 1 is for having coronary heart disease<br/>
'PhysActivity': indicates that 0 is for no physical activity in last 30 days, 1 is for had done physical activity in last 30 days <br/>
'Fruits': indicates that 0 is not consuming at least 1 fruit/day, 1 is for consuming at least 1 fruit/day<br/>
'Veggies': indicates that 0 is for not consuming at least 1 veggies/day, 1 is for consuming at least 1 veggies/day<br/>
'HvyAlcoholConsump': Heavy drinkers (adult men having more than 14 drinks per week and adult women having more than 7 drinks per week) indicates that 0 is for no, 1 is yes <br/>
'AnyHealthcare': indicates that 0 is for no healthcare coverage, 1 is for having health care coverage <br/>
'NoDocbcCost': indicates that 0 is for affording a doctor appointment cause of cost within the last year, 1 is for not being able to afford a doctor appointment cause of cost within the last year<br/>
'GenHlth': personal rating indicates general health scale: scale 1-5 (1 = excellent 2 = very good 3 = good 4 = fair 5 = poor) <br/>
'MentHlth': indicates days thinking about mental health <br/>
'PhysHlth': indicates days thinking about physical health <br/>
'DiffWalk': indicates that 0 is for no difficulty walking down stairs, 1 is for difficulty walking down stairs <br/>
'Sex': indicates that 0 is for female, 1 is for male <br/>
'Age': indicates age in years <br/>
'Education': indicates Education level (EDUCA see codebook) scale 1-6 (1 = Never attended school or only kindergarten; 2 = Grades 1 through 8) <br/>
'Income': indicates Income scale (INCOME2 see codebook) scale 1-8 (1 = less than 10,000; 5 = less than 35,000; 8 = $75,000 or more) <br/>

In [31]:
#dataset with/without diabetes 
diabetes_binary_dataset

Diabetes_binary  HighBP  HighChol  CholCheck   BMI  Smoker  Stroke  \
0                   0.0     1.0       1.0        1.0  40.0     1.0     0.0   
1                   0.0     0.0       0.0        0.0  25.0     1.0     0.0   
2                   0.0     1.0       1.0        1.0  28.0     0.0     0.0   
3                   0.0     1.0       0.0        1.0  27.0     0.0     0.0   
4                   0.0     1.0       1.0        1.0  24.0     0.0     0.0   
...                 ...     ...       ...        ...   ...     ...     ...   
253675              0.0     1.0       1.0        1.0  45.0     0.0     0.0   
253676              1.0     1.0       1.0        1.0  18.0     0.0     0.0   
253677              0.0     0.0       0.0        1.0  28.0     0.0     0.0   
253678              0.0     1.0       0.0        1.0  23.0     0.0     0.0   
253679              1.0     1.0       1.0        1.0  25.0     0.0     0.0   

        HeartDiseaseorAttack  PhysActivity  Fruits  ...  AnyHealthcare  \
0                        0.0           0.0     0.0  ...            1.0   
1                        0.0           1.0     0.0  ...            0.0   
2                        0.0           0.0     1.0  ...            1.0   
3                        0.0           1.0     1.0  ...            1.0   
4                        0.0           1.0     1.0  ...            1.0   
...                      ...           ...     ...  ...            ...   
253675                   0.0           0.0     1.0  ...            1.0   
253676                   0.0           0.0     0.0  ...            1.0   
253677                   0.0           1.0     1.0  ...            1.0   
253678                   0.0           0.0     1.0  ...            1.0   
253679                   1.0           1.0     1.0  ...            1.0   

        NoDocbcCost  GenHlth  MentHlth  PhysHlth  DiffWalk  Sex   Age  \
0               0.0      5.0      18.0      15.0       1.0  0.0   9.0   
1               1.0      3.0       0.0       0.0       0.0  0.0   7.0   
2               1.0      5.0      30.0      30.0       1.0  0.0   9.0   
3               0.0      2.0       0.0       0.0       0.0  0.0  11.0   
4               0.0      2.0       3.0       0.0       0.0  0.0  11.0   
...             ...      ...       ...       ...       ...  ...   ...   
253675          0.0      3.0       0.0       5.0       0.0  1.0   5.0   
253676          0.0      4.0       0.0       0.0       1.0  0.0  11.0   
253677          0.0      1.0       0.0       0.0       0.0  0.0   2.0   
253678          0.0      3.0       0.0       0.0       0.0  1.0   7.0   
253679          0.0      2.0       0.0       0.0       0.0  0.0   9.0   

        Education  Income  
0             4.0     3.0  
1             6.0     1.0  
2             4.0     8.0  
3             3.0     6.0  
4             5.0     4.0  
...           ...     ...  
253675        6.0     7.0  
253676        2.0     4.0  
253677        5.0     2.0  
253678        5.0     1.0  
253679        6.0     2.0  

[253680 rows x 22 columns]

In this dataset, column Diabetes_binary indicates that 0 is for no diabetes and 1 is for diabetes. </br> The columns are the same as the previous dataset.

### OLS Regression  

Both datasets contain the same columns except for their respective first columns (Diabetes_012 and Diabetes_binary). We did OLS regression on the first columns using the 7 numeric features to determine which features would be a valid in modeling the factors that affect diabetes. 

In [32]:
model1 = sm.ols('Diabetes_012 ~  BMI + Income + Education + Age + GenHlth + MentHlth + PhysHlth', data = diabetes_012_dataset)
results = model1.fit()
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:           Diabetes_012   R-squared:                       0.142
Model:                            OLS   Adj. R-squared:                  0.142
Method:                 Least Squares   F-statistic:                     6016.
Date:                Fri, 28 Apr 2023   Prob (F-statistic):               0.00
Time:                        21:32:07   Log-Likelihood:            -2.4933e+05
No. Observations:              253680   AIC:                         4.987e+05
Df Residuals:                  253672   BIC:                         4.988e+05
Df Model:                           7                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     -0.6914      0.011    -62.662      0.0

In [33]:
model2 = sm.ols('Diabetes_binary ~  BMI + Income + Education + Age + GenHlth + MentHlth + PhysHlth', data = diabetes_binary_dataset)
results = model2.fit()
print(results.summary())

                            OLS Regression Results                            
Dep. Variable:        Diabetes_binary   R-squared:                       0.133
Model:                            OLS   Adj. R-squared:                  0.133
Method:                 Least Squares   F-statistic:                     5554.
Date:                Fri, 28 Apr 2023   Prob (F-statistic):               0.00
Time:                        21:32:07   Log-Likelihood:                -72849.
No. Observations:              253680   AIC:                         1.457e+05
Df Residuals:                  253672   BIC:                         1.458e+05
Df Model:                           7                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     -0.3384      0.006    -61.496      0.0

Given that dependent variable Diabetes_binary is binary (0 = no diabete, 1 = diabetes), positive coefficients indicate higher likeliness to have diabetes while negative coefficient has lower likeliness to develop diabetes.

The f-statistic is 5554 and greater than 10, indicating that it's a strong model and has significant explanatory power. 

The p-values of the intercept is 0 which indicate that the data is statistically significant at the 1% level (and 5% level). This means that we reject the null hypothesis for a significance test at 1% and we can be 99% confident that the independent variables has a significant effect on Diabetes_binary. 

The positive coefficients of BMI, age, general health indicate that as these numbers increase, the chances of getting diabetes also increases. This reasonable since older, more overweight (higher BMI), and people who rate their health poorly will have worser health than those who are fit and younger. 

The negative coefficients of income and education indicate that with higher income and better/more years of education, there's less likelihood that one will develop diabetes. More income likely allows easier access to better food and better medical help. 

# Author Contributions Statement

### Bruce: 
I started EDA.ipynb

### Duy: 
I started main.ipynb and did ols regression on variables of interest: diabetes. I explained the OLS regression and its significance of the dataset and model.

### Sam: 

### Donghoon: 